Hermes/appscale/hermes/stats/subscribers/profile.py

```diff
@@ -47,30 +55,380 @@ def receive(self, nodes_stats_dict):
+          new_summary_file_name = '{}.new'.format(summary_file_name)
+          with open(new_summary_file_name, 'w') as new_summary:
+            # Write new header
+            csv.writer(new_summary).writerow(self._summary_columns)
+            # Recover old data
+            new_summary.writelines(old_summary)
+        rename(new_summary_file_name, summary_file_name)
+
+      with open(summary_file_name, 'a') as summary_file:
+        # Append line with the newest summary
+        row = [time.mktime(datetime.utcnow().timetuple())]
+        columns_iterator = self._summary_columns.__iter__()
+        columns_iterator.next()  # Skip timestamp column
+        for service_name in columns_iterator:
+          service_summary = services_summary.get(service_name)
+          if service_summary:
+            row.append(getattr(service_summary, attribute.name))
+          else:
+            row.append('')
+        csv.writer(summary_file).writerow(row)
+
+  def _get_summary_file_name(self, resource_name):
+    name = self._summary_file_name_template.format(resource=resource_name)
+    name = name.replace('_', '-')
+    return path.join(self._directory, name)
 
 
 class ClusterProxiesProfileLog(StatsSubscriber):
 
+  @attr.s(cmp=False, hash=False, slots=True)
+  class ServiceProxySummary(object):
+    """
+    This data structure holds list of most valuable proxy stats attributes.
+    Separate CSV summary file is created for each attribute of this model,
+    so we can easily compare services regarding important properties.
+    """
+    requests_rate = attr.ib(default=0)
+    bytes_in_out = attr.ib(default=0)
+    errors = attr.ib(default=0)
+
   def __init__(self, include_lists=None):
-    # TODO
-    pass
+    """ Initializes profile log for cluster processes stats.
+    Renders header according to include_lists in advance and
+    creates base directory for processes stats profile log.
+    It also reads header of summary file (if it exists) to identify
+    order of columns.
+
+    Args:
+      include_lists: an instance of IncludeLists describing which fields
+        of processes stats should be written to CSV log.
+    """
+    self._header = (
+      ['utc_timestamp']
+       + converter.get_stats_header(proxy_stats.ProxyStats, include_lists)
+    )
+    self._include_lists = include_lists
+    self._directory = path.join(PROFILE_LOG_DIR, 'proxies')
+    self._summary_file_name_template = 'summary-{property}.csv'
+    try:
+      makedirs(self._directory)
+    except OSError as os_error:
+      if os_error.errno == errno.EEXIST and path.isdir(self._directory):
+        pass
+      else:
+        raise
+    self._summary_columns = self._get_summary_columns()
 
   def receive(self, proxies_stats_dict):
-    # TODO
-    pass
+    """ Implements receive method of base class. Saves newly produced
+    cluster proxies stats to a list of CSV files.
+    One detailed file for each proxy on every node (normally one LB node)
```

Comment:
"Three summary files and one detailed file for each proxy on every load balancer node"